### ISSUE TITLE
Data type

### ISSUE THREAD

**Issue by aminamani10** [SimonBlanke/Hyperactive#33]
Hello and tnx for this great library.
in CNN networks, most of the time we used imageDatagenerator which save data and label in a data frame and we have not a separate x and y. this is same for train, validation and test sets. this library can support this? or we need separate data and labels like in examples. if this support dataframeiterator type (output of imageDatagenerator) will be amazing.

**Comment by SimonBlanke** [comment]
Hello @aminamani10,

yes, Hyperactive is very flexible in how you put your model into the objective function. If the image-datagenerator-stuff does not change during the optimization run you can just put it outside the objective function. This way it is created only once instead of each time the objective function is evaluated.

If you post a small script of your model I can show you how to utilize Hyperactive.

**Comment by aminamani10** [comment]
Tnx I could solve that. it will be amazing if you can link this project with pyomo package and adding other metaheuristics algorithms to this package to solve mathematical programming problems too ( and multiobjective mathematical programming problem). all industrial engineers and those who are research in the operation research field will cite your package in their research.

**Comment by aminamani10** [comment]
I have another question too, it takes too much for running this with a deep network, is it possible to run this on a tpu (google colab) too? if it's possible will be great if you put an example code for that. very tnx.

**Comment by SimonBlanke** [comment]
Hello @aminamani10,

I will look into the pyomo package. It will probably be very difficult to link Hyperactive with pyomo, but I could implement some of the algorithms in my package.

It is possible to run the network on a GPU or TPU. Hyperactive does not affect how the model is trained. 
Also: Hyperactive does not slow down the training procedure at all. The computational load from Hyperactive is orders of magnitude smaller than from deep neural network training (except if you go absolutely crazy with the smbo algos). 

If you want to run the code on google colab you can just copy the code from the example into colab and run it. But I will also add some examples in google colab within the next few days.

**Comment by aminamani10** [comment]
i have no problem with running code on colab with GPU, but with TPU it doesn't run. it will be great if u add example for TPU tnx.

**Comment by SimonBlanke** [comment]
@aminamani10 I just ran the tensorflow example from the Hyperactive examples on a TPU in colab and it works without any problems. Could you provide an example I can run, that shows the error in colab?

**Comment by aminamani10** [comment]
unfortunately, I did not save that. but the problem was when the CNN model function wanted to be run inside strategy.scope() function, it did not recognize the params input. it would be great if you add an example for this. tnx
